### PR TITLE
Bring remaining variants autocomplete (and related) JS code from spree_backend

### PR DESCRIPTION
**Pull request by luisramos0**
#### What? Why?

Continues #4595

Bring variant_autocomplete as well as shipments js code to make orders edit page work. We also bring address_states used in the customer details page in the order edit page.

To support this we need to require admin/handlebar_extensions to make translations work inside handlbars templates.
Here spree uses underscore js to call functions like _.find, instead of bringing underscore we require existing (used in the frontoffice) lodash.underscore.js

#### What should we test?
Order edit page and customer details page particularly the auto complete features in them: search variants and search customers.

#### Release notes
Changelog Category: Changed
Bring variant autocomplete code from spree_backend to make ofn independent of Spree.

### CODE REVIEW

File: app/assets/javascripts/admin/spree/orders/shipments.js.erb [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+      url: url
+    }).done(function( msg ) {
+      window.location.reload();
+    }).error(function( msg ) {
+      console.log(msg);
+    });
+  }
+  // $('[data-hook=admin_order_edit_form] a.ship').click(handle_ship_click);
+
+  //handle shipping method edit click
```
Comment (Matt-Yorkley) [inline app/assets/javascripts/admin/spree/orders/shipments.js.erb]: Are all these commented out lines in the originals files as well?

Comment (luisramos0) [reply app/assets/javascripts/admin/spree/orders/shipments.js.erb]: yes, these events are registered in spree_backend. I will remove spree_backend and these comments in https://github.com/openfoodfoundation/openfoodnetwork/pull/4621/files

**Comment by luisramos0** [comment]
Rebased and tested deploying. All good. Ready for testing.

**Comment by lin-d-hop** [comment]
In Edit Products and Customers
Tested a range of combinations - start, mid, end - of name/email/code all relevant fields for the search.
Everything worked a treat.
Ready to go.

**Comment by luisramos0** [comment]
great, thanks for testing!!!

**Pull request merged**